tests/__init__.py

```diff
@@ -0,0 +1,9 @@
+import sys
+import logging
```

Comment:
I was just experimenting with it because we will have conditions running through the same tests where things pass/fail for specific reasons. It becomes less necessary when we split up the tests into separate passing and failing tests, and probably not necessary at all given your other comment to catch specific raised exceptions with their own assertions